### PR TITLE
Fix ticket quantity change after editing date in modal | Closes #2803

### LINKED ISSUE #2803: Date capacity change in edit modal does NOT update related ticket quantity
Currently when changing the datetime capacity inline, it updates the quantity of all the related tickets if needed. Same behavior is needed when it gets changed in edit date modal.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR fixes the issue of date capacity change in edit modal not updating related ticket quantity.

### CODE REVIEW

File: assets/src/domain/eventEditor/ui/datetimes/hooks/useEditDatetimeModal.ts [hunk 1]
```diff
@@ -4,24 +4,42 @@ import { FormProps } from 'react-final-form';
...
+import { useDatetimeItem } from '@edtrServices/apollo';
 
 const useEditDatetimeModal: FormModal = ({ entityId, entityDbId }) => {
 	const { updateEntity } = useDatetimeMutator(entityId);
 	const { closeEditor } = useFormModal();
 
+	const updateRelatedTickets = useUpdateRelatedTickets(entityId);
+	const ticketQuantityForCapacity = useTicketQuantityForCapacity();
+
 	const { name: eventName = '' } = useEvent() || {};
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/datetimes/hooks/useEditDatetimeModal.ts:23]: is it possible to push some of these defaults into `useEvent`, such that the end result would look:
```
	const { name: eventName } = useEvent();
```
?

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/datetimes/hooks/useEditDatetimeModal.ts:23]: Why should `useEvent` return an empty object when there is nothing. It's the responsibility of the consumer to handle `undefined`

File: assets/src/domain/eventEditor/ui/datetimes/hooks/useEditDatetimeModal.ts [hunk 2]
```diff
@@ -4,24 +4,42 @@ import { FormProps } from 'react-final-form';
...
 		closeEditor('editDatetime');
 	}, [closeEditor]);
 
 	const onSubmit = useCallback<FormProps<UpdateDatetimeInput>['onSubmit']>(
 		(fields): void => {
 			updateEntity(fields);
+
+			const capacityChanged = fields?.capacity !== datetime?.capacity;
+
+			if (capacityChanged) {
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/datetimes/hooks/useEditDatetimeModal.ts:37]: :clap: :+1:

File: assets/src/domain/eventEditor/ui/tickets/hooks/useEditTicketModal.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ const useEditTicketModal: FormModal = ({ entityId, entityDbId }) => {
 
 	const ticket = useTicketItem({ id: entityId });
 
-	const { name: eventName = '' } = useEvent() || {};
+	const event = useEvent();
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/tickets/hooks/useEditTicketModal.ts:19]: better than the suggestion :)

**Pull request merged**